### PR TITLE
Add global auth backend for files external

### PR DESCRIPTION
**Pull request by icewind1991**
cc @PVince81 @Xenopathic

### CODE REVIEW

File: apps/files_external/lib/auth/password/globalauth.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * @author Robin Appelman <icewind@owncloud.com>
+ *
+ * @copyright Copyright (c) 2015, ownCloud, Inc.
```
Comment (PVince81) [inline apps/files_external/lib/auth/password/globalauth.php]: you live in the past!

File: apps/files_external/templates/settings.php [hunk 2]
```diff
@@ -164,3 +164,16 @@ function writeParameterInput($parameter, $options, $classes = []) {
 		</p>
 	<?php endif; ?>
 </form>
+<?php if ($_['visibilityType'] === BackendService::VISIBILITY_ADMIN): ?>
```
Comment (PVince81) [inline apps/files_external/templates/settings.php]: Please move this above the storages list. That's also how it was in WND.

**Comment by PVince81** [comment]
Code looks good otherwise, a quick test showed that it works.

Please move the fields above the storages table and this is good to go.

**Comment by PVince81** [comment]
Fixes https://github.com/owncloud/core/issues/21991

**Comment by jmaciasportela** [comment]
In WND, global credentials was available for admin and users:

* Admin: for whole systems
* User: for their personal mount points

Correct me if I'm wrong but here is only for admin.

Other thing that we should think is to turn some of this enhancements to only EE flavour, if we want to have an EE version more valuable.

**Comment by PVince81** [comment]
Right, I forgot to check the personal page, thanks for pointing out @jmaciasportela 

@icewind1991 can you add a set of credentials there too ?

**Comment by icewind1991** [comment]
Now also works for personal mounts

**Comment by PVince81** [comment]
JS unit failures unrelated: #22123

**Comment by PVince81** [comment]
@icewind1991 the password field is missing a "placeholder" attribute

**Comment by PVince81** [comment]
Tested, works :+1:

**Comment by icewind1991** [comment]
> the password field is missing a "placeholder" attribute

For some reason I was convinced that the password obfuscation thing also applied to placeholders...

**Comment by PVince81** [comment]
@icewind1991 no, that bug is only with IE <= 9 I think...

**Comment by icewind1991** [comment]
now with placeholder

> no, that bug is only with IE <= 9 I think...

I think my believe is caused by chrome's over eagerness of autofiling the password

**Comment by PVince81** [comment]
@jmaciasportela second review ?

**Comment by jmaciasportela** [comment]
:+1:

**Comment by jmaciasportela** [comment]
@PVince81 @icewind1991 I figure out that if you have a mount point with global credentials with green status, and change global credentials, mount point still green. It would be good if we recheck the status in all mount points that use global credentials.

**Comment by PVince81** [comment]
@jmaciasportela can you make a ticket ?

CC @SergioBertolinSG @Xenopathic

**Comment by jmaciasportela** [comment]
https://github.com/owncloud/core/issues/22144

**Pull request merged**